c2rust-transpile/src/renamer.rs

```diff
@@ -176,6 +178,14 @@ impl<T: Clone + Eq + Hash> Renamer<T> {
         self.next_fresh += 1;
         self.pick_name(&format!("fresh{}", fresh))
     }
+
+    fn raw_identifier_if_reserved_name(basename: &str) -> Option<String> {
+        if RESERVED_NAMES.contains(&basename) && !RESERVED_PATH_SEGMENT_NAMES.contains(&basename) {
+            Some(format!("r#{}", basename))
+        } else {
+            None
+        }
+    }
```

Comment:
It is a bummer that `RESERVED_NAMES` and `RESERVED_PATH_SEGMENT_NAMES` leak into `Renamer`. I could make the `Renamer` take a closure allowing the detection of reserved names to be configurable.

Reply:
Renamer already takes a reference to `RESERVED_NAMES` as the argument to its `new` constructor, so you could take that reference and stick it inside `Renamer` but I'm not sure that's worth it.

Reply:
Agreed, that was my initial approach (ie treat `reserved_names` as names that should be converted to a raw identifier). However, when I had to add `RESERVED_PATH_SEGMENT_NAMES` there did not seem to be value in keeping the reserved list internally.